src/Http/Controllers/Api/FormController.php

```diff
@@ -0,0 +1,111 @@
+                        'employment',
+                    ]),
+                ],
+                'first_name' => 'required',
+                'last_name' => 'required',
+                'email_address' => 'required|email',
+                'phone' => 'required',
+                'message' => 'nullable|max:3000',
+            ]);
+        } else {
+            $validator = Validator::make($request->all(), [
+                'form_type' => [    'required',
+                    Rule::in([
+                        'on-the-run',
+                        'parties',
+                        'reservations',
+                        'team-building',
+                    ]),
+                ],
+                'first_name' => 'required',
+                'last_name' => 'required',
+                'email' => 'required|email',
+                'phone' => 'required',
+                'message' => 'nullable|max:3000',
+                'company_name' => 'nullable|string',
+            ]);
+        }
+
+        if ($validator->fails()) {
+            $messages = collect($validator->errors()->getMessages())->map(function ($messages, $key) {
+                return ['name' => $key, 'message' => reset($messages)];
+            })->values()->toArray();
+
+            return response(['verifyErrors' => $messages], 500)
+                ->header('AMP-Access-Control-Allow-Source-Origin', $domain);
+        }
+
+        // Don't save spam messages where first name is same as last name
+        if ($request->first_name !== $request->last_name) {
+
+            // Find or create the user
+            $user = app('user')->where('email', $request->email)->first();
+            if (! $user) {
+                $user = app('user')->updateOrCreate(
+                    ['email' => $request->email],
+                    [
+                        'first_name' => $request->first_name,
+                        'last_name' => $request->last_name,
+                    ]
+                );
+            }
+            // Todo: How is this handled now?
+            //$user->locations()->attach($location);
+
+            // Create the contact database entry
+            $contact = new Contact();
+            $contact->user_id = $user->id;
+            $contact->location_id = $request->location_id;
+            $contact->form_type = $request->form_type;
+            $contact->phone = $request->phone;
+            $contact->message = $request->message;
+            if (! in_array($request->form_type, $basic)) {
+                $contact->company_name = $request->company;
+                $contact->fields->requested_date = $request->requested_date;
+                $contact->fields->requested_time = $request->requested_time;
+                $contact->fields->participants = $request->participants;
+            }
+            $contact->save();
+        }
+
+        // Todo: Might be easier to check the config to see if the form is enabled, maybe throw an error if it's not.
```

Comment:
This falls back on using the config to see active forms. Technically they shouldn't have even gotten this far if the form isn't registered as 'active'.